src/ml_gym/util/util.py

```diff
@@ -191,15 +202,26 @@ def update_dataset_details(exp_config: dict) -> DatasetDetails:
                     obj (DatasetDetails): initialized dataset details object.
             """
             try:
-                dataset_splits = {"split_config": exp_config["dataset_iterators"]["config"]["split_configs"], 
-                              "splits_percentage": exp_config["splitted_dataset_iterators"]["config"]["split_configs"]}
+                splits_percentage = []
+                split_subscription = set()
+                find_key = "dataset_iterator"
+                for key in exp_config:
+                    if find_key in key and exp_config[key]["component_type_key"] == "SPLITTED_DATASET_ITERATORS":
```

Comment:
still hardcoded paths / keys here

Reply:
the component_type_key is hardcoded yes, because that is static, it cannot be changed. I have changed find_key to "dataset", so it looks for keys that have the substring dataset and then check for the component key. I do not see any reason for someone to create a dataset iterator component without the word dataset in the key. I can simply just remove the find_key and only search for the component_type_key. What do you think?